VultisigApp/VultisigApp/Services/Solana/SolanaService.swift

```diff
@@ -69,6 +70,59 @@ class SolanaService {
         return blockHash
     }
     
+    // Token metadata https://raw.githubusercontent.com/solana-labs/token-list/main/src/tokens/solana.tokenlist.json
+    // Solana Token   https://solana.com/docs/rpc/http/gettokenaccountsbyowner
+    func fetchTokens(for walletAddress: String) async throws -> [CoinMeta] {
+        let requestBody: [String: Any] = [
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "getTokenAccountsByOwner",
+            "params": [
+                walletAddress,
+                ["programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"], // SPL Token Program ID
+                ["encoding": "jsonParsed"]
+            ]
+        ]
+        do {
+            let data = try await postRequest(with: requestBody)
+            
+            if let jsonResponse = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
```

Comment:
can we use JSONDecoder and Codable please?

Reply:
Yes, sure.